### PR TITLE
Add manifest for org.kde.cantor

### PR DESCRIPTION
**Pull request by flyingcakes85**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - `--share=network` for cantor to connect to internet
  - `--filesystem=host` for load/save/export functionality
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - https://invent.kde.org/education/cantor/-/merge_requests/37

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

<hr>

Notes

Cantor supports multiple backends. This manifest only bundles cantor with defaults, providing KAlgebra.

### CODE REVIEW

**Comment by flyingcakes85** [comment]
bot build org.kde.cantor

**Comment by flyingcakes85** [comment]
bot, build org.kde.cantor

**Comment by flyingcakes85** [comment]
@flathub/kde

File: boost.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "name": "boost",
+    "buildsystem": "simple",
+    "build-commands": [
+        "./bootstrap.sh --prefix=/app --with-libraries=system",
+        "./b2 -j $FLATPAK_BUILDER_N_JOBS install"
+    ],
+    "sources": [
+        {
+            "type": "archive",
+            "url": "https://boostorg.jfrog.io/artifactory/main/release/1.76.0/source/boost_1_76_0.tar.bz2",
```
Comment (travier) [inline boost.json:11]: 1.78.0 is out

File: org.kde.cantor.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    "base-version": "5.15-21.08",
+    "sdk": "org.kde.Sdk",
+    "command": "cantor",
+    "rename-icon": "cantor",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=host"
```
Comment (travier) [inline org.kde.cantor.json:15]: Also likely to need `--device=dri`

**Review by travier — COMMENTED** [review]
Minor comments

File: boost.json (hunk 1 above)
Comment (flyingcakes85) [reply boost.json:11]: I'll update and test. Side note, should this be changed in the master manifest repo too?
https://invent.kde.org/packaging/flatpak-kde-applications/-/blob/master/boost.json#L11

**Comment by flyingcakes85** [comment]
bot, build org.kde.cantor

File: org.kde.cantor.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "base-version": "5.15-21.08",
+    "sdk": "org.kde.Sdk",
+    "command": "cantor",
+    "rename-icon": "cantor",
+    "finish-args": [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--filesystem=host",
```
[the remark this answers is not in the document]
Comment (flyingcakes85) [reply org.kde.cantor.json:15]: Cantor has functionality to load / save worksheets, and export loaded worksheet to different format. Portals should be a safer way, however referring [this page](https://docs.flatpak.org/en/latest/portals-qt.html) it seems like Qt may not be able to work with saving files.

[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.kde.cantor.json:15]: xdg-documents should be enough. host is way too broad.

[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.kde.cantor.json:15]: Qt5 has portal support btw.

[the remark this answers is not in the document]
Comment (flyingcakes85) [reply org.kde.cantor.json:15]: I'll replace it with xdg-documents. :+1:

File: boost.json (hunk 1 above)
Comment (travier) [reply boost.json]: Yes, we need to update it there too. I forgot to do it :).

File: org.kde.cantor.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (travier) [reply org.kde.cantor.json:15]: 👍🏻

**Comment by flyingcakes85** [comment]
bot, build org.kde.cantor

**Comment by flyingcakes85** [comment]
Updated to use `--filesystem=xdg-documents`

**Pull request closed without merging**